imagebuildah/build.go

```diff
@@ -340,6 +340,15 @@ func buildDockerfilesOnce(ctx context.Context, store storage.Store, logger *logr
 
 	warnOnUnsetBuildArgs(logger, mainNode, options.Args)
 
+	// --platform was explicitly selected for this build
+	// so set correct TARGETPLATFORM in args if it is not
+	// already selected by the user.
+	if options.OS != "" && options.Architecture != "" {
+		if _, ok := options.Args["TARGETPLATFORM"]; !ok {
+			options.Args["TARGETPLATFORM"] = options.OS + "/" + options.Architecture
```

Comment:
The `options.SystemContext` will have these values, and the variant value, too, if it's set.

Reply:
Yes looks better, fixed.